src/routes/webhooks.js

```diff
@@ -0,0 +1,140 @@
+import express from 'express';
+import {
+  updateStatusOfferValidator,
+  updateChainIdOfferValidator,
+  updateTokenOfferValidator,
+  updateMinPriceOfferValidator,
+  updateMaxPriceOfferValidator,
+} from '../validators/webhook.validator.js';
+import { validateResult } from '../utils/validators-utils.js';
+import getDBConnection from '../db/conn.js';
+
+const router = express.Router();
+
+/* This is a PUT request that updates max price offer. */
+router.put(
+  '/offer/max-price',
+  updateMaxPriceOfferValidator,
+  async (req, res) => {
+    const validator = validateResult(req, res);
+    const collection = (await getDBConnection(req)).collection('offers');
+    if (validator.length) {
+      return res.status(400).send(validator);
+    }
+    const offer = await collection.findOne({
+      offerId: req.body._idOffer,
+    });
+    if (offer)
+      return res.status(200).send(
+        await collection.updateOne(offer, {
+          $set: {
+            max: req.body._upperLimitFn ? req.body._upperLimitFn : offer.min,
+          },
+        })
+      );
+    res.status(400).send({
+      msg: 'Not found.',
+    });
+  }
+);
+
+/* This is a PUT request that updates min price offer. */
+router.put(
+  '/offer/min-price',
+  updateMinPriceOfferValidator,
+  async (req, res) => {
+    const validator = validateResult(req, res);
+    const collection = (await getDBConnection(req)).collection('offers');
+    if (validator.length) {
+      return res.status(400).send(validator);
+    }
+    const offer = await collection.findOne({
+      offerId: req.body._idOffer,
+    });
+    if (offer)
+      return res.status(200).send(
+        await collection.updateOne(offer, {
+          $set: {
+            min: req.body._lowerLimitFn ? req.body._lowerLimitFn : offer.min,
+          },
+        })
+      );
+    res.status(400).send({
+      msg: 'Not found.',
+    });
+  }
+);
+
+/* This is a PUT request that updates token offer. */
+router.put('/offer/token', updateTokenOfferValidator, async (req, res) => {
+  const validator = validateResult(req, res);
+  const collection = (await getDBConnection(req)).collection('offers');
+  if (validator.length) {
+    return res.status(400).send(validator);
+  }
+  const offer = await collection.findOne({
+    offerId: req.body._idOffer,
+  });
+  if (offer)
+    return res.status(200).send(
+      await collection.updateOne(offer, {
+        $set: {
+          tokenAddress: req.body._token ? req.body._token : offer.tokenAddress,
```

Comment:
@tcoratger This endpoint is listening this [event](https://github.com/grindery-io/Depay-Reality/blob/2b0596f883ac32bf0338a05e7f71e027ab32f8d6/contracts/v-testnet-launch/GrtOffer.sol#L37).

I got confused because I don't know with token is related with token symbol (BNB) or token Address (0x0).
In this case I left as token address to update.

Let me know if its correct.

Reply:
> @tcoratger This endpoint is listening this [event](https://github.com/grindery-io/Depay-Reality/blob/2b0596f883ac32bf0338a05e7f71e027ab32f8d6/contracts/v-testnet-launch/GrtOffer.sol#L37).
> 
> I got confused because I don't know with token is related with token symbol (BNB) or token Address (0x0). In this case I left as token address to update.
> 
> Let me know if its correct.

@juniormp `setTokenOffer(bytes32 offerId, address token)` is a function intended to modify the token relating to the offer previously created, let me explain.

1. Gordon creates an offer corresponding to a given token, in `setOffer`, he fills in the address of the token he wishes to offer (for the moment, we only use the native BNB token in all our offers and this address is therefore address 0). This creation costs him quite a bit of transaction fees and if he just wants to modify the token he offers, it is better that there is a dedicated function.
2. `setTokenOffer` is dedicated to this use: by entering the `offerId` of a previously created offer, he can here simply modify the do token address he is going to offer.
3. But for now this feature is not used because there is only one offer: the one created by Tim to provide testnet BNB to our first users and we don't need to change the token at the moment.